SharedCode/Repository/DB/PokemonRepositoryLocal.cs

```diff
@@ -118,6 +117,54 @@ public async Task<Result<List<PokemonLocal>>> GetPokemonLocalListAsync()
             }
             return Result.Fail<List<PokemonLocal>>("Can't get info from db");
         }
+
+        public void StoreEvolutionChain(EvolutionChainResponse evolutionChain)
+        {
+            var evolutionChainString = Newtonsoft.Json.JsonConvert.SerializeObject(evolutionChain);
+            var localEvolutionChain = new EvolutionChainLocal { Id = evolutionChain.id, Chain = evolutionChainString };
+            DatabaseManager.StoreData(localEvolutionChain, Constants.EvolutionChainTable);
+        }
+
+        public Result<EvolutionChainResponse> GetEvolutionChainByIdFromLocal(int id)
+        {
+            try
+            {
+                var dbResponse = DatabaseManager.GetAllData<EvolutionChainLocal>();
+
+                if (dbResponse.IsFailure)
+                {
+                    return Result.Fail<EvolutionChainResponse>("Could not get data");
+                }
+                var row = dbResponse.Value.Where(r => r.Id == id).First();
+
+                EvolutionChainResponse evolutionChainResponse = JsonConvert.DeserializeObject<EvolutionChainResponse>(row.Chain);
+                var evolutionChainFromLocal = new EvolutionChainResponse { id = row.Id, chain = evolutionChainResponse.chain };
+
+                return Result.Ok<EvolutionChainResponse>(evolutionChainFromLocal);
+            } catch (Exception)
+            {
+                return Result.Fail<EvolutionChainResponse>("Could not get data");
+            }
+        }
+
+        public Result<List<EvolutionChainResponse>> GetAllEvolutionChainFromLocal(int id)
```

Comment:
Same for this function you are getting data so you can declare it in the interface

Reply:
Done.